In [2]:
import statistics as st

In [8]:
  x = [10,20,30,40,35,40,40,50,50,50,50,45,50,60,70,80]

In [9]:
print(st.mean(x))

45


In [10]:
print(st.median(x))

47.5


In [12]:
x[8]

50

In [16]:
x.sort()

In [17]:
print(x)

[10, 20, 30, 35, 40, 40, 40, 45, 50, 50, 50, 50, 50, 60, 70, 80]


In [18]:
 (50 + 45) /2

47.5

In [19]:
# mode 

st.mode(x)
    
    
    

50

# Descriptive statistic
#### 1. __Measure of central tedency__
#### 2. __Measure of dispersion__ (spread)
    

In [35]:
import pandas as pd
import numpy as np
import random
from random import randint as rt


In [44]:
x = np.random.randint(30,35,10)
print(x)
## print random numbers, from 30 to 35 and 10 numbers

[30 30 30 30 30 30 34 32 30 34]


In [45]:
np.mean(x)

31.0

In [48]:
np.median(x)

30.0

In [50]:
st.mode(x)

30

In [52]:
print('variance', np.var(x))

## variance 

variance 2.6


In [55]:
print('Standard deviation', np.std(x))

## It is how the deviate from the mean, (std)

Standard deviation 1.61245154965971


# Quartiles

In [59]:
y = [10,20,30,40,50,60,70,80]
y_ = pd.DataFrame(y)

In [58]:
y_.describe()

0
count   8.000000
mean   45.000000
std    24.494897
min    10.000000
25%    27.500000
50%    45.000000
75%    62.500000
max    80.000000

### Quartile1 = 25% = 27.50
### Quartile2 = 50% = 45
### Quartile3 = 75% = 75

### _What is inter Quartile

#### is __{ Q3 - Q1 }__



# Central Limit Theorem

#### Normal distribution is always normal irrespective to te sample

#### Non-Normal distribution - If sample size is adequate (appr > 30 sample), distribution starts looking normal.

### __ux = u ( Mean of sample mean = population Mean)__
### __6x = 6/sqrt n ( Population std / sqrt of sample)__